src/main/java/uk/gov/companieshouse/insolvency/delta/processor/InsolvencyDeltaProcessor.java

```diff
@@ -47,59 +48,66 @@ public InsolvencyDeltaProcessor(ApiClientService apiClientService,
-
-            final String updatedBy = String.format("%s-%s-%s", receivedTopic, partition, offset);
-
-            internalCompanyInsolvency.getInternalData().setUpdatedBy(updatedBy);
-
-            final String companyNumber = insolvency.getCompanyNumber();
-            logger.trace(String.format("DSND-362: InsolvencyDelta transformed into "
-                    + "InternalCompanyInsolvency: %s", internalCompanyInsolvency));
-            logMap.put("company_number", companyNumber);
-            logger.infoContext(
-                    logContext,
-                    String.format("Process insolvency for company number [%s]", companyNumber),
-                    logMap);
-
-            final ApiResponse<Void> response =
-                    apiClientService.putInsolvency(logContext,
-                            companyNumber,
-                            internalCompanyInsolvency);
-
-            handleResponse(null, HttpStatus.valueOf(response.getStatusCode()), logContext,
-                    "Response from sending insolvency data", logMap);
-        } catch (RetryableErrorException ex) {
-            retryDeltaMessage(chsDelta);
-        } catch (Exception ex) {
-            handleErrorMessage(chsDelta);
-            // send to error topic
+        /** We always receive only one insolvency/charge per delta in a list,
+         * so we only take the first element
+         * CHIPS is not able to send more than one insolvency per delta.
+         **/
+
+        Insolvency insolvency = insolvencyDelta.getInsolvency().get(0);
+
+        InternalCompanyInsolvency internalCompanyInsolvency = transformer.transform(insolvency);
+
+        final String updatedBy = String.format("%s-%s-%s", receivedTopic, partition, offset);
+
+        internalCompanyInsolvency.getInternalData().setUpdatedBy(updatedBy);
+
+        final String companyNumber = insolvency.getCompanyNumber();
+        logger.trace(String.format("DSND-362: InsolvencyDelta transformed into "
+                + "InternalCompanyInsolvency: %s", internalCompanyInsolvency));
+        logMap.put("company_number", companyNumber);
+        logger.infoContext(
+                logContext,
+                String.format("Process insolvency for company number [%s]", companyNumber),
+                logMap);
+
+        final ApiResponse<Void> response =
+                apiClientService.putInsolvency(logContext,
+                        companyNumber,
+                        internalCompanyInsolvency);
+
+        handleResponse(null, HttpStatus.valueOf(response.getStatusCode()), logContext,
+                "Response from sending insolvency data", logMap);
+
+    }
+
+    private InsolvencyDelta mapToInsolvencyDelta(ChsDelta payload)
+            throws NonRetryableErrorException {
+        ObjectMapper mapper = new ObjectMapper();
+        try {
+            return mapper.readValue(payload.getData(),
+                    InsolvencyDelta.class);
+        } catch (JsonProcessingException exception) {
```

Comment:
The readValue method cold throws:
com.fasterxml.jackson.core.JsonProcessingException
com.fasterxml.jackson.databind.JsonMappingException

Please catch also the JsonMappingException

And for both we should throw NonRetryableException.